### PR TITLE
Auth login/signup

### PR DESCRIPTION
**Pull request by MuskanChhatrasal**
Pages to check
login.js
signup.js
authcontext.js
authreducer.js

### CODE REVIEW

File: src/Components/Authentication/signup.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react'
+import { Link } from 'react-router-dom'
+import { useAuth } from '../../Context/AuthContext/authContext'
+import './authentication.css'
+
+const Signup = () => {
+  const [userDetails, setUserDetails] = useState({firstName: '',lastName: '',email: '', password: '', confirmPassword: '', terms: false})
+  const [toggleShowPassword, setToggleShowPassword] = useState(true)
+  const [toggleShowConfirmPass, setToggleShowConfirmPass] = useState(true)
+  const [error, setError] = useState({isError: false, text: ''})
```
Comment (mikesch803) [inline src/Components/Authentication/signup.js:10]: suggestion : you could use the useReducer

File: src/Components/Authentication/signup.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+       setError({isError: false, text: ''})
+      }, 3000)
+
+      return ()=>clearTimeout(timeoutId)
+  },[error])
+
+
+  const submitHandler = (e) =>{
+      e.preventDefault()
+      const passwordValidation = /^(?=.*\d)(?=.*[a-z]).{5,10}$/;
```
Comment (mikesch803) [inline src/Components/Authentication/signup.js:27]: good you are using password validation here

File: src/Components/Authentication/signup.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          setUserDetails({firstName: '', lastName: '', email: '', password: '', confirmPassword: '', terms: false})
+      }
+  }
+  return (
+    <div className="wrapper">
+    {error && <p className='error-text'>{error.text}</p>}
+        <h2>Sign-Up</h2>
+        <form>
+            <div className="input-box">
+                <input type="text" placeholder="Enter your First name" name='firstName' id='name' value={userDetails.fullName} onChange={(e)=>setUserDetails({...userDetails, firstName: e.target.value})} />
```
Comment (mikesch803) [inline src/Components/Authentication/signup.js:58]: suggestion : you are using name attribute, so in setter you could use the [e.target.name] instead firstname, even not only but in input fields

File: src/Components/Authentication/signup.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            <div className="input-box">
+                <input type="text" placeholder="Enter your last name" name='lastName'  value={userDetails.fullName} onChange={(e)=>setUserDetails({...userDetails, lastName: e.target.value})} />
+            </div>
+            <div className="input-box">
+            
+                <input type="text" placeholder="Enter your email"  name='email' id='email' value={userDetails.email} onChange={(e)=>setUserDetails({...userDetails, email: e.target.value})} />
+            
+            </div>
+            <div className="input-box input-icons">
+               {toggleShowPassword ? <i class="fas fa-eye icon" onClick={()=>setToggleShowPassword(!toggleShowPassword)}></i>:<i class="fas fa-eye-slash icon" onClick={()=>setToggleShowPassword(!toggleShowPassword)}></i>}
```
Comment (mikesch803) [inline src/Components/Authentication/signup.js:69]: great, you are using the icons for hide and show password

**Review by mikesch803 — APPROVED** [review]
hey i put some suggestion, apart from that code is easily understandable

**Comment by VLeads** [comment]
Logout functionality is not working. I guess you will implement it in future.

**Review by VLeads — APPROVED** [review]
All code looks good. Nothing extra to add !!

File: src/Components/Authentication/signup.js (hunk 1 above)
Comment (MuskanChhatrasal) [reply src/Components/Authentication/signup.js:10]: Okayy I will replace it with useReducer. Thankyou!!

File: src/Components/Authentication/signup.js (hunk 3 above)
Comment (MuskanChhatrasal) [reply src/Components/Authentication/signup.js:58]: Yes you are right. Will do that. Thankyou for pointing it.

**Comment by MuskanChhatrasal** [comment]
> Logout functionality is not working. I guess you will implement it in future.

It should work. Will see that.

**Pull request merged**